crates/context/primitives/src/client.rs

```diff
@@ -46,37 +59,232 @@ impl ContextClient {
         Ok(Some(context))
     }
 
-    pub async fn get_contexts(&self) -> impl Stream<Item = eyre::Result<ContextId>> {
-        stream::empty()
+    pub async fn get_contexts(
+        &self,
+        start: Option<ContextId>,
+    ) -> impl Stream<Item = eyre::Result<ContextId>> {
+        let datastore = self.datastore.clone();
+
+        try_stream! {
+            let handle = datastore.handle();
```

Comment:
```suggestion
        let handle = datastore.handle();

        try_stream! {
```

I believe this should work